portal/src/graphql/portal/EditSingleSignOnConfigurationScreen.tsx

```diff
@@ -171,7 +188,12 @@ const EditSingleSignOnConfigurationScreen1: React.VFC<{
     }
   }, [isReadyToEdit, onRevealSecrets]);
 
-  if (!isReadyToEdit || form.isLoading || featureConfig.loading) {
+  if (
+    !isReadyToEdit ||
+    form.isLoading ||
+    featureConfig.loading ||
+    isEffectiveSecretLoading
+  ) {
```

Comment:
Maybe we need a new abstraction to handle loading and error in case there are multiple queries being used in a component?

Reply:
Let me try to introduce a new hook or component for this.